text/dependencies/rfcs/0000-dependency-management-phase-two.md

```diff
@@ -0,0 +1,195 @@
+For context, here is an enumeration of the current workflows that run in Github
+Actions within the dep-server repository.
+
+Each dependency has three main workflows:
+- [**Get New
+  Versions**](https://github.com/paketo-buildpacks/dep-server/blob/main/.github/templates/get-new-versions.yml):
+  discover new versions of the dependency by cross checking published versions
+  against a known-versions file on a timer and trigger `Build and Upload`
+  workflow when a new version is discovered.
+
+- [**Build and
+  Upload**](https://github.com/paketo-buildpacks/dep-server/blob/main/.github/templates/build-upload.yml):
+  each new dependency version is compiled or modified from it's upstream
+  source, uploaded to an S3 bucket, and the dependency metadata is gathered and
+  used to trigger the `Test and Upload` workflow.
+
+- [**Test and Upload
+  Metadata**](https://github.com/paketo-buildpacks/dep-server/blob/main/.github/templates/test-upload-metadata.yml):
+  runs tests against the compiled dependency and uploads dependency metadata to
+  S3 bucket if tests pass.
+
+Metadata and the dependencies themselves are stored in S3 buckets and are
+accessible through `https://api.deps.paketo.io/v1/dependency?name=<dependency>`
+through a series of API endpoint routing.
+
+Dependencies are updated in the buildpacks in a workflow that runs
+in the buildpack's Github Actions:
+- **Update Dependencies**: Runs every hour and runs [`jam
+  update-dependencies`](https://github.com/paketo-buildpacks/github-config/blob/f85a65d14fd90f6d63f1af2d408ee38f17ce5c0b/actions/dependency/update/action.yml#L45-L46)
+  against the `buildpack.toml`. If there are new versions available in the
+  dep-server, this workflow will discover them and open a PR to update the
+  `buildpack.toml`.
+
+### New Plan Overview
+The new process for managing dependencies will use the `buildpack.toml` as a
+source of truth for the latest versions we support, instead of using a separate
+`known-versions.json` file. Metadata will also be generated and updated in
+place with a pull request to the `buildpack.toml`, rather than being pushed to
+an intermediary `metadata.json` file. The overall process will be closer to
+what someone would do if they were updating the dependencies themselves, rather
+than with automation. The overall steps will be:
+
+1. Pick up all versions of the dependency, using code from the Phase 1 RFC in
+   `<buildpack>/dependency/retrieval`
+2. Compare the list of all versions with the version constraints and supported
+  versions in the `buildpack.toml` to determine what versions can be updated.
+3. Generate metadata for each of the versions to be added using code from Phase
+   1 in `<buildpack>/dependency/metadata`.
+4. If the `URI` and `SHA256` are missing from the metadata, the code will be
+  compiled with the code from `<buildpack>/dependency/compilation`.
+5. Test the dependency (whether compiled or not) using the test from
+   `<buildpack>/dependency/test`
```

Comment:
If we are using dependencies directly from upstream do we need to test them? In most cases, where we aren't compiling it seems like the workflow should be as simple as generating metadata and PRing `buildpack.toml`. If we need to test anything we could run those tests as a gate on the PR.

Reply:
Let's leave this up to maintainers to decide.